packages/launcher/src/components/launch/Launch.tsx

```diff
@@ -69,36 +70,49 @@ export default function Launch(): ReactElement {
     const connected = await docker.isConnected();
     if (connected) {
       // setPulling(true);
-      setStatus("Pulling Axon image");
-      await docker.pullImage();
-      // setPulling(false);
-      setStatus("Finished pulling.");
-      // image downloaded
-      const containers = await docker.getContainers();
-      if (containers !== null && containers.length > 0) {
-        setStatus("Removing old containers");
-        await docker.reset();
-      }
-      setStatus("Creating container");
-      const container = await docker.createContainer();
-      // setContainer(container);
-      // setContainerReady(true);
-      console.log("Container created.");
-      setStatus("Running container");
-      docker.runContainer(container).then(() => {
-        setStatus("OFF");
-        setActiveContainer(null);
+      dns.resolve("www.google.com", (err) => {
+        if (err) {
+          console.log("No connection, Skipping Pulling Images");
+        } else {
+          console.log("Connected to Internet, Pulling Images");
+          makeNewContainers();
+        }
       });
-      setActiveContainer(container);
-      localhost.waitForStart();
+
     } else {
       setClicked(false);
     }
   };
+
   docker.isConnected().then((value) => {
     setOpen(!value && !clicked);
   });
 
+  const makeNewContainers = async () => {
+    setStatus("Pulling Axon image");
+    await docker.pullImage();
+    // setPulling(false);
+    setStatus("Finished pulling.");
+    // image downloaded
+    const containers = await docker.getContainers();
+    if (containers !== null && containers.length > 0) {
+      setStatus("Removing old containers");
+      await docker.reset();
+    }
+    setStatus("Creating container");
+    const container = await docker.createContainer();
```

Comment:
You could still create a container if the image exists. Try seeing if the image exists?